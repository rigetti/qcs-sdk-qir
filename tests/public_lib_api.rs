use std::fs::read;

use qcs::ExecutionResult;
use qcs_sdk_qir::{
    output::{self, DebugOutputFormat},
    transpile_qir_to_quil,
};

#[test]
fn transpile_qir_to_quil_bell_state() {
    let data = read("tests/fixtures/programs/bell_state.bc").unwrap();
    let output = transpile_qir_to_quil(&data).unwrap();
    insta::assert_snapshot!(output.program.to_string(true));
    insta::assert_display_snapshot!(output.shot_count);
}

#[test]
fn capture_recorded_output_and_convert() {
    let data = read("tests/fixtures/programs/record_output.bc").unwrap();
    let output = transpile_qir_to_quil(&data).unwrap();
    insta::assert_json_snapshot!(&output.recorded_output);

    let debug_format = output::try_format::<DebugOutputFormat>(
        &ExecutionResult::I8(vec![vec![1, 2], vec![2, 4], vec![3, 6]]),
        &output.recorded_output,
    )
    .unwrap();
    insta::assert_snapshot!(debug_format);
}
