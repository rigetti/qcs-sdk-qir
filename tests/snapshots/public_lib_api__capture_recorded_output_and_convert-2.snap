---
source: tests/public_lib_api.rs
expression: debug_format
---
[shot:1 start]
[shot:1 result 1]
[shot:1 tuple_start]
[shot:1 result 1]
[shot:1 result 2]
[shot:1 tuple_end]
[shot:1 array_start]
[shot:1 result 1]
[shot:1 result 1]
[shot:1 array_end]
[shot:1 end]
[shot:2 start]
[shot:2 result 2]
[shot:2 tuple_start]
[shot:2 result 2]
[shot:2 result 4]
[shot:2 tuple_end]
[shot:2 array_start]
[shot:2 result 2]
[shot:2 result 2]
[shot:2 array_end]
[shot:2 end]
[shot:3 start]
[shot:3 result 3]
[shot:3 tuple_start]
[shot:3 result 3]
[shot:3 result 6]
[shot:3 tuple_end]
[shot:3 array_start]
[shot:3 result 3]
[shot:3 result 3]
[shot:3 array_end]
[shot:3 end]
